src/util.h

```diff
@@ -29,6 +29,15 @@
 #include <boost/signals2/signal.hpp>
 #include <boost/thread/exceptions.hpp>
 
+#ifdef DEBUG_ASSERTION
+/// If DEBUG_ASSERTION is enabled this asserts when the predicate is false.
+//  If DEBUG_ASSERTION is disabled and the predicate is false, it executes the execInRelease statements.
+//  Typically, the programmer will error out -- return false, raise an exception, etc in the execInRelease code.
+#define DbgAssert(pred, execInRelease) do { assert(pred); } while(0)
+#else
+#define DbgAssert(pred, execInRelease) do { if (!(pred)) { execInRelease; }} while(0)
```

Comment:
This usage will cause a problem because of the macro ``while``
```
for (int i=0; i<max; i++) {
    DbgAssert(checkSomething(i), continue);
    doSomethingImportant(i);
}
```